tools/fix_mailmap.py

```diff
@@ -0,0 +1,573 @@
+            msg = "   Multiple emails of github-noreply form. "
+            msg += f"Please edit mailmap manually: {emailcount_list}"
+        elif dup_emails_ibm.count(True) == 1:
+            canonical_email = dup_emails[dup_emails_ibm.index(True)]
+            emails_list = ", ".join(f"<{e}>" for e in dup_emails if e != canonical_email)
+            msg = f"    One email is of @ibm.com form: <{canonical_email}>\n"
+            msg += f"    Setting the others ({emails_list}) to match it\n"
+        elif dup_emails_ibm.count(True) > 1:
+            msg = f"   Multiple emails of @ibm.com form. Please edit mailmap manually: {emailcount_list}"
+        else:
+            canonical_email = counted_emails[0][0]
+            msg = (
+                f"    No emails are in mailmap. Choosing <{canonical_email}> as the most common.\n"
+            )
+            msg += f"        {emailcount_list}\n"
+    else:
+        msg = "     No obvious choice for canonical email. Please resolve manually."
+    return canonical_email, msg
+
+
+def _main():
+    exit_code_string = "\n    ".join(f"{x.value}:\t{x.name}" for x in ExitCode)
+    parser = argparse.ArgumentParser(
+        description="Check .mailmap file and repo history for mailmapping problems",
+        epilog="Exit codes:\n    " + exit_code_string,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    git_mailmap_path = get_mailmap_path()
+    parser.add_argument(
+        "--mailmap-file",
+        default=git_mailmap_path,
+        type=Path,
+        help=f"Override mailmap file from {git_mailmap_path}",
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="Quiet mode: suppress non-warning messages",
+    )
+    parser.add_argument(
+        "--color",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+        help="Show colored output",
+    )
+    parser.add_argument(
+        "-s",
+        "--skip-non-canonical-author",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="""Warn on non canonical authors (usually resulting from badly-formatted Co-Authored-By
+        lines). There's usually no way to fix these short of rewriting git history""",
+    )
+
+    parser.add_argument(
+        "-n",
+        "--dry-run",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="""Don't actually change the mailmap, even when this tool considers it safe to do so""",
+    )
+
+    args = parser.parse_args()
+    global mailmap_path, mailmap_body_offset, quiet_mode, color_output
+    mailmap_path = args.mailmap_file
+    quiet_mode = args.quiet
+    color_output = args.color
+
+    # Load the raw author entries
+    raw_authors = get_raw_authors()
+    if len(raw_authors) < 10:
+        print_red("Problem: `get_raw_authors()` returned too few authors")
+        sys.exit(ExitCode.GIT_LOG)
```

Comment:
Why 10?

Reply:
It was just a sanity check but probably if I try this on some of the less-trafficked qiskit repos I'll need to reduce it. Maybe down to just 1.